File: packages/api-core/src/response.ts
import { ResponseMeta } from "./types";

export class Response<T = unknown> {
  readonly status: number;
  private readonly _headers: Headers;
  readonly url: string;
  readonly data?: T;
  readonly raw: unknown;
  readonly meta: ResponseMeta;

  constructor(init: {
    status: number;
    headers: Headers;
    url: string;
    data?: T;
    raw: unknown;
    meta?: ResponseMeta;
  }) {
    this.status = init.status;
    this._headers = init.headers;
    this.url = init.url;
    this.data = init.data;
    this.raw = init.raw;
    this.meta = init.meta ?? {};
    Object.freeze(this.meta);
    Object.freeze(this);
  }

  get headers(): Headers {
    return new Headers(this._headers);
  }
}

File: packages/api-core-python/src/aptx_api_core/_response.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any


@dataclass(frozen=True)
class Response:
    status: int
    headers: dict[str, str] = field(default_factory=dict)
    url: str = ""
    data: Any = None
    raw: Any = None
    meta: dict[str, Any] = field(default_factory=dict)
